### PR TITLE
fix NDFs in host status tests, close host testers

### PR DESCRIPTION
**Pull request by johnathanhowell**
This PR adds more flexibility to the host status tests, fixing an observed NDF. Also properly closes the hosttesters that were previously unclosed.

### CODE REVIEW

File: modules/host/network_test.go [hunk 1]
```diff
@@ -66,37 +66,43 @@ func TestHostWorkingStatus(t *testing.T) {
...
+		// Simulate some setting calls, and see if the host picks up on it.
+		atomic.AddUint64(&ht.host.atomicSettingsCalls, workingStatusThreshold+1)
 
-	// No more settings calls, host should believe it is not working now.
-	time.Sleep(workingStatusFrequency + time.Second)
-	if ht.host.WorkingStatus() != modules.HostWorkingStatusNotWorking {
-		t.Fatal("expected host working status to be modules.HostWorkingStatusNotWorking after waiting workingStatusFrequency with no settings calls")
-	}
+		success := false
+		for start := time.Now(); time.Since(start) < 30*time.Second; time.Sleep(time.Millisecond * 10) {
```
Comment (DavidVorick) [inline modules/host/network_test.go]: can use build.Retry for this loop pattern

File: modules/host/network_test.go [hunk 2]
```diff
@@ -66,37 +66,43 @@ func TestHostWorkingStatus(t *testing.T) {
...
-	time.Sleep(workingStatusFirstCheck + time.Second)
-	if ht.host.WorkingStatus() != modules.HostWorkingStatusNotWorking {
-		t.Fatal("expected host working status to be modules.HostWorkingStatusWorking after incrementing status calls")
-	}
-	time.Sleep(workingStatusFrequency - workingStatusFirstCheck)
-	if ht.host.WorkingStatus() != modules.HostWorkingStatusWorking {
-		t.Fatal("expected host working status to be modules.HostWorkingStatusWorking after incrementing status calls")
+		// make no settins calls, host should flip back to NotWorking
+		success = false
+		for start := time.Now(); time.Since(start) < 30*time.Second; time.Sleep(time.Millisecond * 10) {
```
Comment (DavidVorick) [inline modules/host/network_test.go]: ditto

File: modules/host/network_test.go (hunk 1 above)
Comment (lukechampine) [inline modules/host/network_test.go]: looks like `build.Retry` never made it into master?

Comment (lukechampine) [inline modules/host/network_test.go]: ah, `build.Retry` was added in the #1758. I can push that commit to master if you want. Even if we merge #1758, `build.Retry` won't make it into master until we merge the gateway-proto branch.

Comment (johnathanhowell) [inline modules/host/network_test.go]: ah, I forgot that we exported that pattern

Comment (DavidVorick) [inline modules/host/network_test.go]: Ok, sounds good to me. It's useful 😛

**Pull request merged**